src/BenchmarksApps/TodosApi/ProducesResponseTypeMetadata.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.AspNetCore.Http.Metadata;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http;
+
+internal sealed class ProducesResponseTypeMetadata : IProducesResponseTypeMetadata
+{
+    private readonly IEnumerable<string> _contentTypes;
+
+    public ProducesResponseTypeMetadata(int statusCode)
+        : this(typeof(void), statusCode, Enumerable.Empty<string>())
+    {
+    }
+
+    public ProducesResponseTypeMetadata(Type type, int statusCode)
+    {
+        Type = type ?? throw new ArgumentNullException(nameof(type));
+        StatusCode = statusCode;
+        _contentTypes = Enumerable.Empty<string>();
+    }
+
+    public ProducesResponseTypeMetadata(int statusCode, string contentType)
+    {
+        StatusCode = statusCode;
+        MediaTypeHeaderValue.Parse(contentType);
+        _contentTypes = GetContentTypes(contentType, Array.Empty<string>());
+    }
+
+    public ProducesResponseTypeMetadata(Type type, int statusCode, string contentType, params string[] additionalContentTypes)
+    {
+        ArgumentNullException.ThrowIfNull(contentType);
+
+        Type = type ?? throw new ArgumentNullException(nameof(type));
+        StatusCode = statusCode;
+
+        MediaTypeHeaderValue.Parse(contentType);
+        for (var i = 0; i < additionalContentTypes.Length; i++)
+        {
+            MediaTypeHeaderValue.Parse(additionalContentTypes[i]);
+        }
+
+        _contentTypes = GetContentTypes(contentType, additionalContentTypes);
+    }
+
+    // Only for internal use where validation is unnecessary.
```

Comment:
This is a weird comment. What does "Only for internal" mean? This is `private`.

Reply:
This type was copied from dotnet/aspnetcore as it's internal there.

Reply:
Removed it